src/cli.cc

```
@@ -395,6 +394,25 @@ ot::status ot::run(const ot::options& opt)
 	rc = process(reader, &writer, opt);
 	if (rc == ot::st::ok)
 		rc = temporary_output.commit();
-
 	return rc;
 }
+
+ot::status ot::run(const ot::options& opt)
+{
+	if (opt.print_help) {
+		fputs(help_message, stdout);
+		return st::ok;
+	}
+
+	ot::status global_rc = st::ok;
+	for (const auto& [path_in, path_out] : opt.paths_in_out) {
+		ot::status rc = run_single(opt, path_in, path_out);
+		if (global_rc == st::ok)
+			global_rc = rc;
+		else if (!rc.message.empty())
+			/* Report errors other than the one returned. */
+			fprintf(stderr, "error: %s\n", rc.message.c_str());
```

That thing means all but the first messages will be displayed in order, and then the first message will be printed last. That’s super weird. Don’t bother with the special case and print all the messages here, then remove the print from main. If you need to expose run_single so that tests won’t break, feel free to do so.

Concerning the error message to return, since its only caller will eventually only check if it’s OK or not, you can return a generic failure as I suggested.

---

Done. Don't need to expose `run_single`, but do need to print the error message in `main` when parsing fails.